### ISSUE TITLE
i18n doesn't Work

### ISSUE THREAD

**Issue by bcw222** [EveSunMaple/Frosti#57]
**Title:** 🛠️ Bug Report: i18n doesn't Work

**Date:** 2024/12/31

**Affected Version:** Tested on 2.9.0

**Description:**
i18n doesn't work and all strings rendered with label. prefix

**Steps to Reproduce:**

1. Init the repo following the readme;
2. Run `pnpm astro dev` or deploy the repo by any other way.

**Expected Result:**
Seeing categoryCard tagCard and so

**Actual Result:**
See label.categoryCard label.tagCard and so.

**Severity:** Medium

**Priority:** High

**Workaround:** Replace all `t("label.` with `t("`.

**Comment by EveSunMaple** [comment]
Hi @bcw222,  

Thank you for reporting this issue! I’ve checked the setup and translation files, and everything seems to be working correctly on my side. Let me address some points:  

1. **Key Structure:** The prefix `label.` is intentional, as it organizes keys within the translation file. For example, `t("label.categoryCard")` should correctly return "Categories".  

  File: public\locales\en\translation.json
  ```json
  {
    "label": {
      "noTag": "No tags assigned",
      "tagCard": "Tags",
      "tagPage": "Tag",
      "noCategory": "No categories assigned",
      "categoryCard": "Categories",
      "categoryPage": "Category",
      "link": "Link: ",
      "prevPage": "Recent posts",
      "nextPage": "Older posts",
      "wordCount": "words",
      "readTime": "minutes",
      "share": "Share",
      "shareCard": "Share this article",
      "close": "Close"
    }
  }
  ```
   
   > Of course, I am not a native English speaker and using the name 'label' may cause misunderstandings, so if you have a better suggestion, that would be even better.
2. **Potential Misunderstanding:** If you’re directly using `t("categoryCard")` without the `label.` prefix, it will not resolve to the correct key. You can fix this by using `t("label.categoryCard")` instead.  

3. **Workaround Comment:** Removing `label.` might work for your specific use case but would break the structure of the translation file and could cause issues elsewhere.  

4. **Reproducibility:** I wasn’t able to reproduce the issue with the provided steps. If the issue persists, could you share a minimal reproducible example or more details about your environment?  
  ![QQ_1735688077708](https://github.com/user-attachments/assets/36cbd5ca-8a52-4fd2-9892-fccc4f5eed28)


Thanks again for your input, and let me know how I can assist further! 😊

**Comment by bcw222** [comment]
I know it's strange and it's exactly what confused me. It seems that in my scenario even strings with `label.` prefix cannot be handled with i18next correctly.
I'm not familiar with astro and i18next so maybe you can tell me some way to convey more related infomation for the reproducibility.
![image](https://github.com/user-attachments/assets/8fb6d583-1190-44c1-8a18-58b36f883651)

> I'm not a native speaker either and just use English as a tradition on GitHub, so feel free for that.

**Comment by bcw222** [comment]
Even the direct deploy on vercel doesn't work as well.
<https://frosti-iota.vercel.app/>

**Comment by EveSunMaple** [comment]
Alright, I’ve taken a closer look, and it seems the issue is indeed occurring.  

![QQ_1735715884516](https://github.com/user-attachments/assets/f0f2b2b5-4b55-45c7-a510-376fe926b511)  

I checked your repository, `github/bcw222` (a forked repo without new commits), and attempted to reproduce the issue. Strangely enough, when I opened it on Stackblitz, everything built without any problems.  

![QQ_1735716071580](https://github.com/user-attachments/assets/89b7e675-594a-48be-9285-b53f6f201326)  

I also reviewed the language settings on the URL you provided, and they seem to be correct.  

However, when I cloned my own project, the issue finally appeared…  

Interestingly, this problem only occurs on the Vercel deployment. On both local and Stackblitz environments, everything works perfectly.  

In any case, I’ve successfully reproduced the issue now. I’ll work on fixing it as soon as possible—it’s certainly a strange one.

**Comment by bcw222** [comment]
Well, on my local environment this problem also occours, on macOS and Windows. If you need more information, I'm willing to work with you.

**Comment by EveSunMaple** [comment]
Alright, I understand. It’s just that I’m finding this a bit unbelievable: it’s the exact same project, even the packages are identical, so why is it producing different results… I can’t even reproduce the error locally myself.

In the past 30 minutes, I’ve been trying to upgrade this project to Astro v5. I ran out of ideas since there haven’t been any new commits between the working version and the one with errors, so I thought syncing to the latest version might resolve the issue. However, Astro v5 introduced significant changes, and I couldn’t complete the upgrade in one go. After using `npx @astrojs/upgrade`, my project ended up in a disastrous state, and it didn’t even provide a specific address for the issue. (By the way, after I removed the math rendering plugin, the problem surprisingly disappeared. But theoretically, this issue shouldn’t be related to remark or similar plugins.)

In summary, I’m a bit lost right now. This problem feels like a ghost that appeared out of nowhere. Next, I’ll try to upgrade the theme to Astro v5 as soon as possible, hoping it resolves the issue.

That said, I’m a student and will have to go to school soon, so I might not be able to address this until the weekend. I sincerely apologize for the inconvenience this problem has caused you. For now, the solution I can offer is to replace the original i18n content with plain strings.

Of course, I’m happy to collaborate with you to resolve this issue.

Thank you~

**Comment by bcw222** [comment]
I noticed you mentioned you couldn't reproduce the error locally. I've added Docker dev container support to help with this and published several images: 
`registry.cn-hangzhou.aliyuncs.com/bcws/frosti:dev-22-2.9.0`, 
`registry.cn-hangzhou.aliyuncs.com/bcws/frosti:dev-20-2.9.0`, and 
`registry.cn-hangzhou.aliyuncs.com/bcws/frosti:dev-18-2.9.0`.

Interestingly, while setting up the Docker dev container, I discovered the root cause which might explain why you couldn't reproduce the issue: the Node.js version. I apologize for not mentioning this earlier, but I'm currently using v22.12.0. The issue appears to be resolved when testing with Node.js versions 20 and 18. It might be beneficial to add a note about this in the README and consider upgrading your project soon.

Thanks for your time and patience with this issue. Please feel free to close it now, or when the project has been upgraded to support Node v22.

**Comment by EveSunMaple** [comment]
﻿
Thank you for identifying the root cause and providing Docker dev container images. It’s clear now that the issue is related to the Node.js version. I appreciate your efforts in helping debug this problem and your suggestion to upgrade the project for compatibility with Node.js v22.
﻿
I’ve updated the README to include a note about the Node.js version compatibility. These changes should help others avoid similar issues in the future.  
﻿
I’ll work on upgrading the project to fully support Node.js v22 as soon as possible, but it may take some time due to my current schedule. In the meantime, I’ll ensure there’s a clear workaround for users facing this issue.  
﻿
Thank you again for your patience and support! If there are any further updates or feedback, feel free to share. 😊

**Comment by EveSunMaple** [comment]
Fixed #61

**Comment by EveSunMaple** [comment]
new version https://github.com/EveSunMaple/Frosti/releases/tag/v2.9.1

**Comment by EveSunMaple** [comment]
If you have additional questions, you can always open this issue
